src/app/loadout-builder/loadout-builder-reducer.ts

```diff
@@ -205,15 +208,45 @@ export type LoadoutBuilderAction =
   | { type: 'updateSubclassSocketOverrides'; socketOverrides: { [socketIndex: number]: number } }
   | { type: 'removeSingleSubclassSocketOverride'; plug: PluggableInventoryItemDefinition }
   | { type: 'lockExotic'; lockedExoticHash: number }
-  | { type: 'removeLockedExotic' }
+  | { type: 'removeLockedExotic' };
+
+type LoadoutBuilderUIAction =
   | { type: 'openModPicker'; plugCategoryHashWhitelist?: number[] }
   | { type: 'closeModPicker' }
   | { type: 'openCompareDrawer'; set: ArmorSet }
   | { type: 'closeCompareDrawer' };
 
+export type LoadoutBuilderAction =
+  | LoadoutBuilderConfigAction
+  | LoadoutBuilderUIAction
+  | { type: 'undo' }
+  | { type: 'redo' };
+
+function lbUIReducer(state: LoadoutBuilderUI, action: LoadoutBuilderUIAction) {
```

Comment:
Neat, I like having these be split up.

A la Redux, if you have the reducers do `default: return state`, you can just call all of them for all actions and maybe simplify your combined reducer. But it's not that big a deal here.